server-common/src/main/java/org/apache/cassandra/sidecar/common/server/dns/DnsResolver.java

```diff
@@ -26,20 +28,38 @@
  */
 public interface DnsResolver
 {
-    DnsResolver DEFAULT = new DnsResolver()
-    {
-        @Override
-        public String resolve(String hostname) throws UnknownHostException
+    enum DnsResolverEnum implements DnsResolver {
+        @JsonProperty("default")
+        DEFAULT("default")
         {
-            return InetAddress.getByName(hostname).getHostAddress();
+            @Override
+            public String reverseResolve(String address) throws UnknownHostException
+            {
+                return InetAddress.getByName(address).getHostName();
+            }
+        },
+        @JsonProperty("resolveToIp")
+        RESOLVE_TO_IP("resolveToIp")
+        {
+            @Override
+            public String reverseResolve(String address) throws UnknownHostException
+            {
+                return InetAddress.getByName(address).getHostAddress();
+            }
+        };
+
+        final String name;
+
+        DnsResolverEnum(String name)
+        {
+            this.name = name;
         }
 
-        @Override
-        public String reverseResolve(String address) throws UnknownHostException
+        public String toString()
```

Comment:
can we add the override annotation?
```suggestion
        @Override
        public String toString()
```

Reply:
Updated